File: executable/python/Blackbody-heatmap.py
import plotly.plotly as py
from plotly.graph_objs import *

py.sign_in('test-runner', '9h29fe3l0x')

import numpy as np

size = 50
z = np.zeros((size, size)).tolist()
for r in range(1, size + 1):
    for c in range(1, size + 1):
        z[r-1][c-1] = np.sqrt(r*c/float(size**2))

data = Data([
    Heatmap(
        z=z,
        scl='Blackbody'
    )
])
layout = Layout(
    title='Blackbody'
)
fig = Figure(data=data, layout=layout)

plot_url = py.plot(fig, filename='Blackbody-heatmap')

File: executable/r/Blackbody-heatmap.r
library(plotly)

p <- plotly(username='test-runner', key='9h29fe3l0x')

size <- 50
z <- matrix(0, size, size)
for(r in 1:size) {
    for(c in 1:size) {
        z[r, c] <- sqrt(r*c/size^2)
    }
}

data <- list(
  list(
    z = z, 
    scl = "Blackbody", 
    type = "heatmap"
  )
)
layout <- list(title = "Blackbody")

response <- p$plotly(data, kwargs=list(layout=layout, filename="Blackbody-heatmap", fileopt="overwrite"))
url <- response$url
filename <- response$filename